### ISSUE TITLE
Help: ...Unable to access the containers I have in the container network

### ISSUE THREAD

**Issue by Aman4672** [qdm12/gluetun#213]
**TLDR**: *Unable to access the containers i have in the network container:gluetun via ip or localhost via. I can access them fine when they are not inside that network. ports are 9117 and 8080 respectfully an i have tried changing QBs port to 8081.*

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What VPN service provider are you using?

    - [x] PIA
    - [ ] Mullvad
    - [ ] Windscribe
    - [ ] Surfshark
    - [ ] Cyberghost

3. What's the version of the program?

    **See the line at the top of your logs**

    `Running version latest built on 2020-03-13T01:30:06Z (commit d0f678c)`

Running version latest built on 2020-07-23T22:16:53Z (commit 43b9975

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```log
Running version latest built on 2020-07-23T22:16:53Z (commit 43b9975),

2020-07-29T01:45:18.146Z	INFO	OpenVPN version: 2.4.9,
2020-07-29T01:45:18.147Z	INFO	Unbound version: 1.10.1,
2020-07-29T01:45:18.148Z	INFO	IPtables version: v1.8.4,
2020-07-29T01:45:18.165Z	INFO	TinyProxy version: 1.10.0,
2020-07-29T01:45:18.166Z	INFO	ShadowSocks version: 3.3.4,
2020-07-29T01:45:18.167Z	INFO	Settings summary below:,
OpenVPN settings:,
|--User: [redacted],
|--Password: [redacted],
|--Verbosity level: 1,
|--Run as root: no,
|--Private Internet Access settings:,
 |--Network protocol: udp,
 |--Region: us east,
 |--Encryption preset: strong,
 |--Port forwarding: off,
System settings:,
|--User ID: 1000,
|--Group ID: 1000,
|--Timezone: america/new_york,
|--IP Status filepath: /ip,
DNS over TLS settings:,
 |--DNS over TLS provider:,
  |--cloudflare,
 |--Caching: enabled,
 |--Block malicious: enabled,
 |--Block surveillance: disabled,
 |--Block ads: disabled,
 |--Allowed hostnames:,
  |--,
2020-07-29T01:45:18.146Z	INFO	OpenVPN version: 2.4.9,
2020-07-29T01:45:18.147Z	INFO	Unbound version: 1.10.1,
2020-07-29T01:45:18.148Z	INFO	IPtables version: v1.8.4,
2020-07-29T01:45:18.165Z	INFO	TinyProxy version: 1.10.0,
2020-07-29T01:45:18.166Z	INFO	ShadowSocks version: 3.3.4,
2020-07-29T01:45:18.167Z	INFO	Settings summary below:,
OpenVPN settings:,
|--User: [redacted],
|--Password: [redacted],
|--Verbosity level: 1,
|--Run as root: no,
|--Private Internet Access settings:,
 |--Network protocol: udp,
 |--Region: us east,
 |--Encryption preset: strong,
 |--Port forwarding: off,
System settings:,
|--User ID: 1000,
|--Group ID: 1000,
|--Timezone: america/new_york,
|--IP Status filepath: /ip,
DNS over TLS settings:,
 |--DNS over TLS provider:,
  |--cloudflare,
 |--Caching: enabled,
 |--Block malicious: enabled,
 |--Block surveillance: disabled,
 |--Block ads: disabled,
 |--Allowed hostnames:,
  |--,
 
 |--Private addresses:,
  |--127.0.0.1/8,
  |--172.16.0.0/12,
  |--10.0.0.0/8,
  |--192.168.0.0/16,
  |--169.254.0.0/16,
  |--::ffff:0:0/96,
  |--fe80::/10,
  |--fc00::/7,
 |--Verbosity level: 1/5,
 |--IPv6 resolution: disabled,
 |--Update: every 24h0m0s,
 |--Validation log level: 0/2,
 |--Verbosity details level: 0/4,
 |--Keep nameserver (disabled blocking): no,
 |--VPN input ports: ,
Public IP check period: 12h0m0s,
,
ShadowSocks settings: disabled,
TinyProxy settings: disabled,
 |--Allowed subnets: 10.0.0.0/24,
Firewall settings:,
2020-07-29T01:45:18.167Z	INFO	routing: default route found: interface eth0, gateway 172.17.0.1,
2020-07-29T01:45:18.167Z	INFO	routing: local subnet found: 172.17.0.0/16,
2020-07-29T01:45:18.167Z	INFO	openvpn configurator: checking for device /dev/net/tun,
2020-07-29T01:45:18.167Z	WARN	TUN device is not available: open /dev/net/tun: no such file or directory,
2020-07-29T01:45:18.167Z	INFO	openvpn configurator: creating /dev/net/tun,
2020-07-29T01:45:18.167Z	INFO	firewall: enabling...,
2020-07-29T01:45:18.178Z	INFO	firewall: setting allowed subnets through firewall...,
2020-07-29T01:45:18.178Z	INFO	firewall: enabled successfully,
2020-07-29T01:45:18.167Z	INFO	Launching standard output merger,
2020-07-29T01:45:18.180Z	INFO	dns configurator: using DNS address 1.1.1.1 internally,
2020-07-29T01:45:18.180Z	INFO	dns configurator: using DNS address 1.1.1.1 system wide,
2020-07-29T01:45:18.180Z	INFO	dns over tls: falling back on plaintext DNS at address 1.1.1.1,
2020-07-29T01:45:18.179Z	INFO	routing: adding 10.0.0.0/24 as route via 172.17.0.1 eth0,
2020-07-29T01:45:18.181Z	INFO	firewall: setting VPN connections through firewall...,
2020-07-29T01:45:18.180Z	INFO	http server: listening on 0.0.0.0:8000,
2020-07-29T01:45:18.207Z	INFO	openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020,
2020-07-29T01:45:18.207Z	INFO	openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10,
2020-07-29T01:45:18.209Z	INFO	openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]194.59.251.152:1197,
2020-07-29T01:45:18.205Z	INFO	openvpn configurator: starting openvpn,
2020-07-29T01:45:18.209Z	INFO	openvpn: CRL: loaded 1 CRLs from file [[INLINE]],
2020-07-29T01:45:18.209Z	INFO	openvpn: UDP link local: (not bound),
2020-07-29T01:45:18.209Z	INFO	openvpn: UDP link remote: [AF_INET]194.59.251.152:1197,
2020-07-29T01:45:18.365Z	WARN	openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1570', remote='link-mtu 1542',
2020-07-29T01:45:18.365Z	WARN	openvpn: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC',
2020-07-29T01:45:18.365Z	WARN	openvpn: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1',
2020-07-29T01:45:18.365Z	WARN	openvpn: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128',
2020-07-29T01:45:18.365Z	INFO	openvpn: [d50fb3dbd6549023c1aeaf8facbafc21] Peer Connection Initiated with [AF_INET]194.59.251.152:1197,
2020-07-29T01:45:24.422Z	INFO	openvpn: TUN/TAP device tun0 opened,
2020-07-29T01:45:24.423Z	INFO	openvpn: /sbin/ip link set dev tun0 up mtu 1500,
2020-07-29T01:45:24.423Z	INFO	openvpn: /sbin/ip addr add dev tun0 local 10.83.10.6 peer 10.83.10.5,
2020-07-29T01:45:24.427Z	INFO	openvpn: UID set to nonrootuser,
2020-07-29T01:45:24.427Z	INFO	openvpn: Initialization Sequence Completed,
2020-07-29T01:45:24.427Z	INFO	dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated,
2020-07-29T01:45:24.427Z	INFO	routing: default route found: interface eth0, gateway 172.17.0.1,
2020-07-29T01:45:24.427Z	INFO	Gateway VPN IP address: 194.59.251.152,
2020-07-29T01:45:24.658Z	INFO	ip getter: Public IP address is 194.59.251.152,
2020-07-29T01:45:24.737Z	INFO	dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated,
2020-07-29T01:45:24.755Z	INFO	dns configurator: generating Unbound configuration,
2020-07-29T01:45:24.896Z	INFO	dns configurator: 60220 hostnames blocked overall,
2020-07-29T01:45:24.896Z	INFO	dns configurator: 2523 IP addresses blocked overall,
2020-07-29T01:45:24.925Z	INFO	dns configurator: starting unbound,
2020-07-29T01:45:24.925Z	INFO	dns configurator: using DNS address 127.0.0.1 internally,
2020-07-29T01:45:24.925Z	INFO	dns configurator: using DNS address 127.0.0.1 system wide,
2020-07-29T01:45:25.077Z	INFO	unbound: init module 0: validator,
2020-07-29T01:45:25.078Z	INFO	unbound: init module 1: iterator,
2020-07-29T01:45:25.293Z	INFO	unbound: generate keytag query _ta-4a5c-4f66. NULL IN,
2020-07-29T01:45:25.086Z	INFO	unbound: start of service (unbound 1.10.1).
```

Configuration file:

```yml

version: "2.4"
services:
  gluetun:
    image: qmcgaw/private-internet-access
    container_name: gluetun
    cap_add:
      - NET_ADMIN
    network_mode: bridge
    ports:
      - 8888:8888/tcp # Tinyproxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
      - 8001:8000/tcp # Built-in HTTP control server
      - 9117:9117/tcp # Jackett
      - 6881:6881     #
      - 6881:6881/udp # QBT
      - 8080:8080     #
      
    # command:
    environment:
      # More variables are available, see the readme table
      - VPNSP=private internet access

      # Timezone for accurate logs times
      - TZ=america/new_york

      # All VPN providers
      - USER=nicetry

      # All VPN providers but Mullvad
      - PASSWORD=not this time

      # All VPN providers but Mullvad
      - REGION=us east

      # Allow for example your LAN, set to: 192.168.1.0/24
      - EXTRA_SUBNETS=10.0.0.0/24
    volumes:
      - /media/sauron/Erebor/Smaug/Gluetun:/config
    restart: always
```

Host OS:
Ubuntu

```yml
---
version: "2.1"
services:
  jackett:
    image: linuxserver/jackett
    container_name: jackett
    network_mode: "container:gluetun"
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=america/new_york
      - AUTO_UPDATE=true #optional
      - RUN_OPTS=run options here #optional
    volumes:
      - /media/sauron/Erebor/Smaug/Jackett:/config
      - /media/sauron/Erebor/Smaug/Qbittorrent/Downloads:/downloads
    restart: unless-stopped
```

```yml
---
version: "2.1"
services:
  qbittorrent:
    image: linuxserver/qbittorrent
    container_name: qbittorrent
    network_mode: "container:gluetun"
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=america/new_york
      - UMASK_SET=022
      - WEBUI_PORT=8080
    volumes:
      - /media/sauron/Erebor/Smaug/Qbittorrent:/config
      - /media/sauron/Erebor/Smaug/Qbittorrent/Downloads:/downloads
    restart: unless-stopped
```

**Comment by Frepke** [comment]
Does it help when you put /tcp behind your port declaration in the Gluetun compose-file?

**Comment by qdm12** [comment]
The configuration you shown should work. The only thing bugging me is perhaps the docker-compose version which is a bit old, try with `3.7` for example? You could also try

```sh
docker run -it --rm -e LISTENING_PORT=6881 --network="container:gluetun" qmcgaw/port-checker
```

without running qbittorrent and try accessing http://localhost:6881

**Comment by Aman4672** [comment]
> 
> 
> Does it help when you put /tcp behind your port declaration in the Gluetun compose-file?

nope didn't change anything



> 
> 
> The configuration you shown should work. The only thing bugging me is perhaps the docker-compose version which is a bit old, try with `3.7` for example? You could also try
> 
> ```shell
> docker run -it --rm -e LISTENING_PORT=6881 --network="container:gluetun" qmcgaw/port-checker
> ```
> 
> without running qbittorrent and try accessing http://localhost:6881

I cant believe i forgot i did the version change. Portainer uses only compose v2 so i changed the yaml to fit, Headed to work now but i will try what you said, and using the requested version of compose later.

**Comment by Aman4672** [comment]
Yea i am just an idiot sorry. Issue closed

**Comment by qdm12** [comment]
Well using Docker is generally not for idiots. Even less so for taking the time creating an issue on Github. And even less so for closing the issue 😉 Don't be so harsh on yourself 😄  Enjoy tunneling then 👍

**Comment by Frepke** [comment]
> 
> 
> Yea i am just an idiot sorry. Issue closed

I think one mistake doesn't make you an idiot 😄

**Comment by Rustymage** [comment]
Just wanted to say, this helped me fix my issue :)

**Comment by qdm12** [comment]
@Rustymage You were using docker-compose 2 because of portainer as well? It might be useful to write it down somewhere too if it's a commonly encountered issue.

**Comment by Rustymage** [comment]
> @Rustymage You were using docker-compose 2 because of portainer as well? It might be useful to write it down somewhere too if it's a commonly encountered issue.

For me it was the port 9117 in the gluetun compose and the error in my jackett compose, rather than the version.

Although I can see how it would catch people out.
